validator/validator.go

```diff
@@ -16,10 +17,28 @@ var (
 	v    *validator.Validate
 )
 
+func validateResourceBounds(fl validator.FieldLevel) bool {
+	req, ok := fl.Parent().Interface().(rpcv8.Transaction)
+	if !ok {
+		return false
+	}
+
+	if (req.Version.String() == "0x3" || req.Version.String() == "0x100000000000000000000000000000003") &&
+		(req.ResourceBounds == nil || len(*req.ResourceBounds) != 3) {
+		return false
+	}
+	return true
+}
+
 // Validator returns a singleton that can be used to validate various objects
 func Validator() *validator.Validate {
 	once.Do(func() {
 		v = validator.New()
+
+		if err := v.RegisterValidation("resource_bounds_required", validateResourceBounds); err != nil {
+			log.Fatalf("failed to register validation: %v", err)
```

Comment:
I don't think we need to use 'log' here
```suggestion
			panic("failed to register validation: " + err.Error())
```

Reply:
Why do you think `panic` is better than `log.Fatalf` in this case? Just curious. cc @AnkushinDaniil

Reply:
I don't see a big difference between `panic` and `log.Fatalf` here, but we use `panic` in this package for all other cases that we don't expect, so it seems that we can make the code simpler by using `panic`